src/tpm2_wrap.c

```diff
@@ -7029,837 +7029,323 @@ int wolfTPM2_GetPolicyDigest(TPM_HANDLE sessionHandle, byte* policyDigest,
+    rc = wc_HashGetDigestSize(hashType);
+    if (rc < 0)
+        return rc;
+    *digestSz = rc;
 
-        /* set hash alg */
-        policySignedIn->auth.signature.any.hashAlg = pcrAlg;
-
-        /* set auth signature */
-        switch (authKey->pub.publicArea.type) {
-            case TPM_ALG_ECC:
-                policySignedIn->auth.signature.ecdsa.signatureR.size =
-                    policySignedSigSz / 2;
-                XMEMCPY(policySignedIn->auth.signature.ecdsa.signatureR.buffer,
-                    policySignedSig,
-                    policySignedIn->auth.signature.ecdsa.signatureR.size);
-
-                policySignedIn->auth.signature.ecdsa.signatureS.size =
-                    policySignedSigSz / 2;
-                XMEMCPY(policySignedIn->auth.signature.ecdsa.signatureS.buffer,
-                    policySignedSig + policySignedSigSz / 2,
-                    policySignedIn->auth.signature.ecdsa.signatureS.size);
-                break;
-            case TPM_ALG_RSA:
-                policySignedIn->auth.signature.rsassa.sig.size =
-                    policySignedSigSz;
-                XMEMCPY(policySignedIn->auth.signature.rsassa.sig.buffer,
-                    policySignedSig, policySignedSigSz);
-                break;
-            default:
-                rc = BAD_FUNC_ARG;
-                break;
-        }
+    rc = wc_HashInit(&hash_ctx, hashType);
+    if (rc != 0)
+        return rc;
 
-        if (rc == 0) {
-            rc = TPM2_PolicySigned(policySignedIn, policySignedOut);
-        }
+    /* policyDigestOld */
+    if (rc == 0 && inSz > 0) {
+        rc = wc_HashUpdate(&hash_ctx, hashType, digest, inSz);
     }
-
-    /* get the nvAttributes */
+    /* Command Code */
     if (rc == 0) {
-        parent->hndl = TPM_RH_OWNER;
-
-        rc = wolfTPM2_GetNvAttributesTemplate(parent->hndl, &nvAttributes);
+        val = TPM2_Packet_SwapU32(TPM_CC_PolicyAuthorize);
+        rc = wc_HashUpdate(&hash_ctx, hashType, (byte*)&val, sizeof(val));
     }
-
-    /* create the policy digest nv space */
+    /* Public Name Compute */
     if (rc == 0) {
-        rc = wolfTPM2_NVCreateAuth(dev, parent, nv, policyDigestNvIndex,
-            nvAttributes, sizeof(TPM_ALG_ID), NULL, 0);
-
-        if (rc == TPM_RC_NV_DEFINED)
-            rc = 0;
+        TPM2B_NAME name;
+        rc = wolfTPM2_ComputeName(pub, &name);
+        if (rc == 0) {
+            rc = wc_HashUpdate(&hash_ctx, hashType, name.name, name.size);
+        }
     }
-
-    /* authorize the policy using the nv contents */
     if (rc == 0) {
-        /* set session */
-        policyAuthNVIn->policySession = session->handle.hndl;
-        TPM2_SetSessionAuth(dev->session);
-
-        /* set nvIndex */
-        policyAuthNVIn->nvIndex = policyDigestNvIndex;
-
-        /* set authHandle */
-        policyAuthNVIn->authHandle = parent->hndl;
-
-        rc = TPM2_PolicyAuthorizeNV(policyAuthNVIn);
+        rc = wc_HashFinal(&hash_ctx, hashType, digest);
```

Comment:
this also overwrites digest, this might be okay since it's the new policyDigest anyway right?